lib/urireader.cpp

```diff
@@ -3,11 +3,63 @@
 #include "timing.h"
 #include "urireader.h"
 #include "util.h"
+#include "encode.h"
 #include <sys/mman.h>
 #include <sys/stat.h>
+#include <cstdlib>
+#include <ctime>
 
 namespace HTTP{
 
+  // Helper for Date header
+  std::string dateToString() {
```

Comment:
Instead of creating a local function here, I'd prefer a new function in `lib/timing.cpp`, in similar style to `Util::getUTCString(uint64_t epoch){` (perhaps `Util::getDateString(uint64_t epoch)`? And just like that other function, make 0 the default argument value and use the current time when it's zero.)